src/actions/motion.ts

```diff
@@ -615,6 +615,12 @@ class MarkMovementBOL extends BaseMovement {
       await ensureEditorIsActive(mark.document);
     }
 
+    const editor = vscode.window.activeTextEditor;
+    if (!editor) {
+      throw VimError.fromCode(ErrorCode.MarkNotSet);
+    }
+    vimState.editor = editor;
+
     return TextEditor.getFirstNonWhitespaceCharOnLine(vimState.document, mark.position.line);
```

Comment:
Is it sufficient to just use `activeTextEditor.document` here?

Reply:
Unfortunately, if L622(assignment to `vimState`) is removed, it doesn't work.
It would be because `vimState` remains to be old Editor when `Position` is returned.

Do you have any idea?

Reply:
Yeah, I figured as much. I'll have to think about it some more - our current architecture is just very clumsy for actions which relate to multiple editors, since `VimState` is tightly coupled with a `TextEditor`, and an `Action` is run with respect to a given `VimState`.